pkg/apis/core/validation/cloudprofile.go

```diff
@@ -410,3 +412,101 @@ func validateRegions(regions []core.Region, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateBastion(spec *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs     field.ErrorList
+		machineArch *string
+	)
+
+	if spec.Bastion == nil {
+		return allErrs
+	}
+
+	if spec.Bastion.MachineType != nil {
+		var validationErrors field.ErrorList
+		machineArch, validationErrors = validateBastionMachineType(spec.Bastion.MachineType, spec.MachineTypes, fldPath.Child("machineType"))
+		allErrs = append(allErrs, validationErrors...)
+	}
+
+	if spec.Bastion.MachineImage != nil {
+		allErrs = append(allErrs, validateBastionImage(spec.Bastion.MachineImage, spec.MachineImages, machineArch, fldPath.Child("machineImage"))...)
+	}
+
+	return allErrs
+}
+
+func validateBastionMachineType(bastionMachine *core.BastionMachineType, machineTypes []core.MachineType, fldPath *field.Path) (*string, field.ErrorList) {
+	machineIndex := slices.IndexFunc(machineTypes, func(machine core.MachineType) bool {
+		return machine.Name == bastionMachine.Name
```

Comment:
Should this not be `type` or `machineType` instead of `machine`?
The same is also true for `bastionMachine` vs. `bastionMachineType`.

Reply:
The name `type` won't be possible since it's a golang keyword, but you are right.
`machineType` and `bastionMachineType` are more precise.